File: src/Paths/Node.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package Paths;

import file_io.ReadFunctions;

/**
 *
 * @author Kilian
 */
public class Node {

    public Node(ReadFunctions rf){
        int memAdress = rf.readInt();
        int zero = rf.readInt();
        int areaID = rf.readShort();
        int nodeID = rf.readShort();
        int unknown = rf.readInt();
        int always7FFE = rf.readShort();
        int linkID = rf.readShort();
        int posX = rf.readShort();
        int posY = rf.readShort();
        int posZ = rf.readShort();
        byte pathWidth = rf.readByte();
        byte pathType = rf.readByte();
        int flags = rf.readInt();
        System.out.println("--------------Node---------------");
        System.out.println("Mem Adress: " + memAdress);
        System.out.println("Zero: " + zero);
        System.out.println("AreaID: " + areaID);
        System.out.println("NodeID: " + nodeID);
        System.out.println("unknown: " + unknown);
        System.out.println("Always7FFE: " + always7FFE);
        System.out.println("LinkID: " + linkID);
        System.out.println("PosX: " + (float)(posX/8));
        System.out.println("PosY: " + (float)(posY/8));
        System.out.println("PosZ: " + (float)(posZ/128));
        System.out.println("PathWidth: " + pathWidth);
        System.out.println("PathType: " + pathType);
        System.out.println("Flags: " + flags);
    }


}

File: src/main/java/nl/shadowlink/tools/shadowlib/paths/Node.kt
package nl.shadowlink.tools.shadowlib.paths

import nl.shadowlink.tools.io.ReadFunctions

/**
 * @author Shadow-Link
 */
class Node(rf: ReadFunctions) {
    val memAdress = rf.readInt()
    val zero = rf.readInt()
    val areaID = rf.readShort()
    val nodeID = rf.readShort()
    val unknown = rf.readInt()
    val always7FFE = rf.readShort()
    val linkID = rf.readShort()
    val posX = rf.readShort()
    val posY = rf.readShort()
    val posZ = rf.readShort()
    val pathWidth = rf.readByte()
    val pathType = rf.readByte()
    val flags = rf.readInt()

    override fun toString(): String {
        return StringBuilder("--------------Node---------------")
            .append("Mem Adress: $memAdress")
            .append("Zero: $zero")
            .append("AreaID: $areaID")
            .append("NodeID: $nodeID")
            .append("unknown: $unknown")
            .append("Always7FFE: $always7FFE")
            .append("LinkID: $linkID")
            .append("PosX: " + (posX / 8).toFloat())
            .append("PosY: " + (posY / 8).toFloat())
            .append("PosZ: " + (posZ / 128).toFloat())
            .append("PathWidth: $pathWidth")
            .append("PathType: $pathType")
            .append("Flags: $flags")
            .toString()
    }
}